otj-metrics-core/src/main/java/com/opentable/metrics/SortedEntry.java

```diff
@@ -0,0 +1,76 @@
+package com.opentable.metrics;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ComparisonChain;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+
+import com.opentable.metrics.http.HealthController;
+
+public class SortedEntry implements Comparable<SortedEntry> {
+    final String name;
+    final HealthCheck.Result result;
+
+    public static Map<SortedEntry, HealthCheck.Result> render(boolean all, Map<String, HealthCheck.Result> raw) {
+        Map<SortedEntry, HealthCheck.Result> rendered = new TreeMap<>();
+        raw.forEach((name, result) -> {
+            rendered.put(new SortedEntry(ClassUtils.getAbbreviatedName(name, 20), result), result);
+        });
+
+        if (!all && !rendered.isEmpty()) {
+            HealthCheck.Result worstResult = rendered.keySet().iterator().next().getResult();
+            if (!worstResult.isHealthy()) {
+                rendered.keySet().removeIf(e -> e.getResult().isHealthy());
```

Comment:
Couldn't we just not add healthy items to the map in the first place? Why add and then remove?